django/core/management/commands/runserver.py

```diff
@@ -168,23 +143,52 @@ def inner_run(self, *args, **options):
                 handler,
                 ipv6=self.use_ipv6,
                 threading=threading,
+                on_bind=self.on_bind,
                 server_cls=self.server_cls,
             )
         except OSError as e:
             # Use helpful error messages instead of ugly tracebacks.
             ERRORS = {
-                errno.EACCES: "You don't have permission to access that port.",
-                errno.EADDRINUSE: "That port is already in use.",
-                errno.EADDRNOTAVAIL: "That IP address can't be assigned to.",
+                errno.EACCES: f"You don't have permission to access port {self.port}.",
+                errno.EADDRINUSE: f"Port {self.port} is already in use.",
+                errno.EADDRNOTAVAIL: f"IP address {self.addr!r} can't be assigned to.",
             }
             try:
                 error_text = ERRORS[e.errno]
             except KeyError:
                 error_text = e
-            self.stderr.write("Error: %s" % error_text)
+            self.stderr.write(f"Error: {error_text}")
             # Need to use an OS exit because sys.exit doesn't work in a thread
             os._exit(1)
         except KeyboardInterrupt:
             if shutdown_message:
                 self.stdout.write(shutdown_message)
             sys.exit(0)
+
+    def on_bind(self, server_port):
```

Comment:
I wonder if it would make sense to also pass in `server_name` here which is also set by `http.server` and use that instead of `self.addr` -- might get rid of the ifs below.

Reply:
Can you explain how `httpd.server_name` can be used instead of `self.addr` ? I guess I did not understand this part. Because `httpd.server_name` has its value as `localhost` or the host name of my system, depending on the IP addr provided.

Reply:
Ah okay, I thought it would be the resolved/supplied address